sql/src/main/java/org/opensearch/sql/sql/domain/SQLQueryRequest.java

```diff
@@ -64,33 +64,38 @@ public class SQLQueryRequest {
   @Accessors(fluent = true)
   private boolean sanitize = true;
 
+  private String cursor = "";
   /**
    * Constructor of SQLQueryRequest that passes request params.
    */
   public SQLQueryRequest(
-      JSONObject jsonContent, String query, String path, Map<String, String> params) {
+      JSONObject jsonContent, String query, String path, Map<String, String> params, String cursor) {
     this.jsonContent = jsonContent;
     this.query = query;
     this.path = path;
     this.params = params;
     this.format = getFormat(params);
     this.sanitize = shouldSanitize(params);
+    // TODO hack
+    this.cursor = cursor == null? "" : cursor;
```

Comment:
Using optional?

Reply:
Decided to rework cursor [de]serialization and [de]compression in a follow-up task.